### PR TITLE
refactor(tree2): tree builds can be centralized in ModularChangeFamily changesets

### PR DESCRIPTION
**Pull request by yann-achard-MS**
Key changes:
* At the changeset layer, tree builds _can_ now be centralized at the root of ModularChangeFamily changesets. This is however not mandatory.
* At the delta layer, tree builds _can_ now be centralized at the root of deltas. This is however not mandatory.
* Sequence field now leverages the above capabilities.

Note that in this first version, the builds are atomized.

Punted:
* Using a non atomized in-memory and on-wire representation ([task](https://dev.azure.com/fluidframework/internal/_workitems/edit/6307))
* Migrating other field kinds to using the centralized builds ([task](https://dev.azure.com/fluidframework/internal/_workitems/edit/6308))

### CODE REVIEW

File: experimental/dds/tree2/src/feature-libraries/sequence-field/invert.ts [hunk 1]
```diff
@@ -121,7 +122,7 @@ function invertMark<TNodeChange>(
 							id: mark.id,
 							cellId: outputId,
 							count: mark.count,
-							detachIdOverride: mark.cellId,
+							detachIdOverride: getInputCellId(mark, revision, revisionMetadata),
```
Comment (alex-pardes) [inline experimental/dds/tree2/src/feature-libraries/sequence-field/invert.ts:125]: Is this related to the rest of the PR?

Comment (yann-achard-MS) [reply experimental/dds/tree2/src/feature-libraries/sequence-field/invert.ts:125]: It was needed when I was trying to avoid using AttachAndDetach marks for cases where the attach is a new insert.
As we dicussed I can't make that work yet because we need to keep the insert effect (so we can compare its ID to the the ID of the cell).
That said, this change is still good. I've just pushed a pair of tests that fail if getInputCellId is not used.

File: experimental/dds/tree2/src/feature-libraries/modular-schema/modularChangeCodecs.ts [hunk 2]
```diff
@@ -146,19 +154,43 @@ function makeV0Codec(
 		return decodedChange;
 	}
 
+	function encodeBuilds(builds: ModularChangeset["builds"]): EncodedBuilds | undefined {
+		if (builds === undefined) {
+			return undefined;
+		}
+		const encoded: EncodedBuilds = nestedMapToFlatList(builds).map(([r, i, t]) =>
+			r !== undefined ? [r, i, t] : [i, t],
```
Comment (alex-pardes) [inline experimental/dds/tree2/src/feature-libraries/modular-schema/modularChangeCodecs.ts:162]: This feels like a potentially unnecessary optimization. Maybe at least add a quick comment either here or in the format file about this optimization.

**Review by alex-pardes — APPROVED** [review]
